middleware/kubernetes/handler.go

```diff
@@ -115,6 +121,10 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 	return dns.RcodeSuccess, nil
 }
 
+func NewCNAME(name string, target string, ttl uint32) *dns.CNAME {
```

Comment:
You could use this in `OverrideMsg` as well. But this function duplicates some of the things that already exists in `etcd/msg/service.go` where `NewXXX` is defined for a bunch a RR types. Sad thing is they are methods on *Service. Probably needs to be factor out and put into `dnsutil`. Can you add a  TODO here that this needs to be done?

Also the function does not need to be expored